### PR TITLE
Deprecate dead code in QueryJoinParser and remove internal usage

### PR DESCRIPTION
**Pull request by antograssiot**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | 
| License       | MIT
| Doc PR        | 

Follows https://github.com/api-platform/core/pull/2267

### CODE REVIEW

File: tests/Bridge/Doctrine/Orm/Util/QueryJoinParserTest.php [hunk 1]
```diff
@@ -43,30 +43,50 @@ public function testGetClassMetadataFromJoinAlias()
         $this->assertEquals($metadata, $classMetadata->reveal());
     }
 
+    /**
+     * @group legacy
+     * @expectedDeprecation The use of "ApiPlatform\Core\Bridge\Doctrine\Orm\Util\QueryJoinParser::getJoinRelationship()" is deprecated since 2.3 and will be removed in 3.0. Use "Doctrine\ORM\Query\Expr\Join::getJoin()" directly instead.
```
Comment (teohhanhui) [inline tests/Bridge/Doctrine/Orm/Util/QueryJoinParserTest.php]: Actually I'm not sure why we add double quotes where they're unnecessary? (And it makes the message harder to read...)

Comment (antograssiot) [reply tests/Bridge/Doctrine/Orm/Util/QueryJoinParserTest.php]: I can remove them later, when I'll get a computer, it is just a copy/paste from an other error in the codebase somewhere though :stuck_out_tongue:

Comment (antograssiot) [reply tests/Bridge/Doctrine/Orm/Util/QueryJoinParserTest.php]: I've checked and this is also how it is done in symfony, so I guess we should stick to this convention...?

**Comment by dunglas** [comment]
Thanks! Good one!

File: tests/Bridge/Doctrine/Orm/Util/QueryJoinParserTest.php (hunk 1 above)
Comment (teohhanhui) [reply tests/Bridge/Doctrine/Orm/Util/QueryJoinParserTest.php]: Oh, okay...

**Pull request merged**